atlp-rwanda/cabal-bn-be
# #181339537 room booking

### What does this PR do?

Users should be able to book a room in an accommodation facility

### Description of Task to be completed?
 
Users of Barefoot Nomad should be able to book accommodation facilities on the platform depending on what the "supplier(s)/host(s)" have to offer

As a "requester", I should be able to book a room in an accommodation facility.
so that I can be assured of accommodation at my destination.

### How should this be manually tested?

- Clone this [repo](https://github.com/atlp-rwanda/cabal-bn-be.git)

- Check out this branch `feature/#181339537-room-booking`

- Type **`yarn install`** to install dependencies, and **`yarn run dbmigrate:all`** to reinitialize the database.

- Use the postman to test the following endpoints

   POST

                 http://localhost:3000/api/v1/rooms/{roomId}/booking
         

   GET

                 http://localhost:3000/api/v1/rooms/{roomId}/booking

   GET

                 http://localhost:3000/api/v1/rooms/{roomId}/booking/{bookingId}
                
   PATCH

                 http://localhost:3000/api/v1/rooms/{roomId}/booking/{bookingId}

   DELETE

                 http://localhost:3000/api/v1/rooms/{roomId}/booking/{bookingId}

- Documentation: [Room Booking](https://staging-bare-feature-18-xwyu4v.herokuapp.com/api-docs/#/Booking/book%20a%20a%20specific%20room)

         


### What are the relevant pivotal tracker stories?

[#2555280](https://www.pivotaltracker.com/n/projects/2555280)


###Screenshots

<img width="1437" alt="Screen Shot 2022-04-01 at 12 15 57" src="https://user-images.githubusercontent.com/60324751/161244524-edcc9244-6359-4e58-bb7a-5fea360cf9a1.png">
